nservicebus/azure-service-bus/addressing-logic.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Azure Service Bus Transport Addressing Logic
+summary: Azure ServiceBus Transport Addressing Logic
+tags:
+- Azure
+- Cloud
+- Configuration
+---
+
+## Physical Addressing Logic
+
+One of the responsibilities of the transport is determining the names and physical location of entities in the underlying physical infrastructure. This is achieved by turning logical endpoint names into physical addresses of the Azure Service Bus entities, which is called *Physical Addressing Logic*.
+
+### Version 6 and below
+
+In Versions 6 and below, the *Physical Addressing Logic* implementation made a number of implicit assumptions regarding names, such as length limitations, legal characters, differences between paths and names, etc. There was no way to explicitly control those settings, and over time subtle variations started to show up.
+
+For example `Mixed` namespaces allowed paths and names to have up to 290 characters, while `Messaging` namespaces are capped at 260 characters. Moreover, customers were sometimes forced to game the system for valid reasons. For example they needed to figure out how to embed slashes (`'/'`) in endpoint names to create subfolders in the Azure Service Bus namespace registration system.
+
+To mitigate these changes, the transport started to expose certain checks as lambda expressions, which became collectively known as the [Naming Conventions](/nservicebus/azure-service-bus/naming-conventions.md#version-6-and-below-naming-conventions)
+
+### Version 7
+
+In version 7, and above, the configuration API allows to modify all behaviours and assumptions related to the addressing logic. This article will mainly focus on describing the different aspects of the addressing logic and will show how to replace parts if further changes should occur, for a full list of out of the box options refer to [Full Configuration API](/nservicebus/azure-service-bus/configuration/configuration.md).
+
+### Addressing Aspects
+
+The following aspects are found in the addressing logic:
+
+ * Validation: Determines how entity names and paths are validated.
+ * Sanitization: Determines how invalid entity names are cleaned up.
+ * Individualization: Determines how entity names are modified when used in different consumption modes.
+ * NamespacePartitioning: Determines how entities are partitioned across namespaces.
+ * Composition: Determines how entities are hierarchically composed inside a single namespace.
```

Comment:
Composition is dealing with 2 types of entities that have a path (queues and topics). Other entities (subscriptions and rules) don't have paths and composition is not relevant. Perhaps the description should be focused on "path composition"?